nixos/modules/services/web-apps/mediawiki.nix

```diff
@@ -87,7 +96,8 @@ let
 
       ## Database settings
       $wgDBtype = "${cfg.database.type}";
-      $wgDBserver = "${cfg.database.host}:${if cfg.database.socket != null then cfg.database.socket else toString cfg.database.port}";
+      $wgDBserver = "${dbAddr}";
+      $wgDBport = "${toString cfg.database.port}";
```

Comment:
With postgresql this field is always mandatory and ignored for mysql: https://www.mediawiki.org/wiki/Manual:$wgDBport